### PR TITLE
chore: Replace EFS standalone resources with EFS module

### LINKED ISSUE #1080: Replace EFS resources with EFS module
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

- Replace EFS resources used in examples with the EFS module

#### Describe the solution you would like

- EFS module is used in place of the standalone EFS resources

#### Describe alternatives you have considered

- None

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?

- Replace EFS standalone resources with EFS module

### Motivation

- Resolves #1080

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/stateful/main.tf [hunk 1]
```diff
@@ -216,3 +209,16 @@ resource "kubernetes_storage_class" "storage_class" {
     type      = "gp3"
   }
 }
+
+resource "kubernetes_storage_class_v1" "efs" {
```
Comment (Zvikan) [inline examples/stateful/main.tf:213]: Just curious, what's the reason we need this now ?

**Review by Zvikan — APPROVED** [review]
LGTM, I have 1 knowledge question regards `kubernetes_storage_class_v1`. Thank you!

File: examples/stateful/main.tf (hunk 1 above)
Comment (bryantbiggs) [reply examples/stateful/main.tf:213]: it looks like we didn't fill out the example completely/thoroughly (and there is still plenty of room for improvements) but we have the [EFS CSI addon](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/0848d87e4ab52f76dfff2e3d0c6d64d196b6e877/examples/stateful/main.tf#L73) enabled but we never demonstrated how it maps to the StorageClass like we do between the EBS CSI driver and its associated StorageClass resource. So this just demonstrates one way of consuming the EFS system using the dynamic provisioner method 

tl;dr - do we need it - the addon can be enabled without it but users wouldn't be able to utilize an EFS file share (sort of like creating a cluster without a data plane - you can, it just isn't very useful nor how its usually used)
